server/utils/referrals/caseListUtils.ts

```diff
@@ -52,6 +61,57 @@ export default class CaseListUtils {
     ])
   }
 
+  static pagination(
+    path: Request['path'],
+    queryParamsExcludingPage: Array<QueryParam>,
+    pageNumber: Paginated<ReferralSummary>['pageNumber'],
+    totalPages: Paginated<ReferralSummary>['totalPages'],
+  ): GovukFrontendPaginationWithItems {
+    const oneIndexedPageNumber = pageNumber + 1
+
+    const pagination: GovukFrontendPaginationWithItems = { items: [] }
+
+    const numberPageLinkNumbers = Array.from(
+      new Set([1, oneIndexedPageNumber - 1, oneIndexedPageNumber, oneIndexedPageNumber + 1, totalPages]),
+    ).filter((number: Paginated<ReferralSummary>['pageNumber']) => number > 0 && number <= totalPages)
+
+    numberPageLinkNumbers.forEach((numberPageLinkNumber, numberPageLinkNumberIndex) => {
+      if (
+        numberPageLinkNumber !== 1 &&
+        numberPageLinkNumber > numberPageLinkNumbers[numberPageLinkNumberIndex - 1] + 1
+      ) {
+        pagination.items.push({ ellipsis: true, href: '' })
+      }
+
+      const item: GovukFrontendPaginationItem = {
+        href: CaseListUtils.paginationLinkUrl(path, queryParamsExcludingPage, numberPageLinkNumber),
+        number: numberPageLinkNumber.toString(),
+      }
+
+      if (numberPageLinkNumber === oneIndexedPageNumber) {
+        item.current = true
+      }
+
+      pagination.items.push(item)
+    })
+
+    if (oneIndexedPageNumber > 1) {
+      pagination.previous = {
+        href: CaseListUtils.paginationLinkUrl(path, queryParamsExcludingPage, oneIndexedPageNumber - 1),
+        text: 'Previous',
```

Comment:
This text is not needed: the default has "page" visually hidden, so this removes that